### PR TITLE
Separate unit tests and integration tests

### PR DESCRIPTION
**Pull request by NicolasMahe**
This PR separate integration test from unit test on CircleCI.
The unit test runs in only 8sec 🍾 
The code coverage is now only based on the unit tests. (that's why there is -5% of code coverage)

The ultimate thing to do about integration test is to transform it to a workflow that uses an application and a service to test many situation once.
I recommend to keep integration test until then.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -64,9 +78,9 @@ jobs:
       - <<: *restore_go_mod_cache
       - run: go mod download
       - <<: *save_go_mod_cache
-      - <<: *restore_go_cache
+      # - <<: *restore_go_cache
```
Comment (krhubert) [inline .circleci/config.yml]: why this is not restored anymore?

Comment (krhubert) [inline .circleci/config.yml]: why go cache is commented here?

Comment (NicolasMahe) [reply .circleci/config.yml]: Nice catch

File: .circleci/config.yml [hunk 2]
```diff
@@ -43,17 +43,31 @@ jobs:
...
       - run: docker swarm init
-      - run: docker build -t sleep docker-images/sleep/
-      - run: docker build -t http-server docker-images/http-server/
       - <<: *restore_go_mod_cache
       - run: go mod download
       - <<: *save_go_mod_cache
       - run: mkdir -p ~/.mesg
       - <<: *restore_go_cache
-      - run: go test -v -timeout 300s -p 1 -tags=integration -coverprofile=coverage.txt ./...
+      - run: go test -v -timeout 300s -coverprofile=coverage.txt ./...
```
Comment (antho1404) [inline .circleci/config.yml]: maybe we could even lower the timeout (but this might be a test for another time)

Comment (NicolasMahe) [reply .circleci/config.yml]: yes we can. all the test takes less that 8sec :)

Comment (NicolasMahe) [reply .circleci/config.yml]: but i'm lazy to change it...

**Pull request merged**